spyder/plugins/completion/kite/client.py

```diff
@@ -170,4 +172,16 @@ def perform_request(self, req_id, method, params):
                 converter = getattr(self, converter_name)
                 if response is not None:
                     response = converter(response)
-        self.sig_response_ready.emit(req_id, response or {})
+        if not isinstance(response, dict):
+            QMessageBox.critical(
+                self, _('Kite error'),
```

Comment:
I think that the `QMessageBox.critical` needs a `QWidget` as parent. Since the `KiteClient` is only a QObject maybe that's the reason why some test are failing?

Reply:
I think to avoid this problem we should show the message only if `running_under_pytest()`  is False.